gtsam/hybrid/HybridGaussianFactor.cpp

```diff
@@ -56,7 +56,7 @@ HybridGaussianFactor::Factors augment(
                           const HybridGaussianFactor::sharedFactor &gf) {
     auto jf = std::dynamic_pointer_cast<JacobianFactor>(gf);
     if (!jf) return gf;
-    // If the log_normalizer is 0, do nothing
+    // If the value is 0, do nothing
     if (values(assignment) == 0.0) return gf;
```

Comment:
You are calling the potentially expensive `values(assignment)` twice, so please assign it to a variable.  But why unzip the tree and then call `values()` at all. Can't you just apply on the original tree and get the value for free?

A more problematic issue is that you take the square root. But is this wrong now? Should we not do sqrt(2*value) ? Because the "hidden" value c will be squared and halved: `value == 0.5*c^2`.

Reply:
1. So I need to unzip to normalize the values (so that we don't have negatives, else the sqrt(value) gives a nan). This brings up an interesting point though: should we normalize for the user or should the user pre-normalize? I felt like us doing it makes sense, since the sqrt to hide the variable is something we need to do and the user shouldn't have to worry about those semantics. Thoughts?

2. We can do 2*value, but that is going to have to also change in various other places like `HybridGaussianConditional::likelihood`. I think what you suggest is the better thing, keeping in alignment with my previous point.